packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -0,0 +1,237 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { concatStyleSets, IStyle, ITextFieldStyles, mergeStyles, TextField, useTheme } from '@fluentui/react';
+import React, { useState } from 'react';
+import { useLocale } from '../../localization';
+import { containerStyles, digitStyles, rowStyles, subStyles, textFieldStyles } from './Dialpad.styles';
+
+/**
+ * Strings of {@link Dialpad} that can be overridden.
+ *
+ * @public
+ */
+export interface DialpadStrings {
+  errorText: string;
+  defaultText: string;
+}
+
+/**
+ * Props for {@link Dialpad} component.
+ *
+ * @public
+ */
+export interface DialpadProps {
+  strings?: DialpadStrings;
+  containerStyles?: IStyle;
+  rowStyles?: IStyle;
+  textFieldStyles?: Partial<ITextFieldStyles>;
+  digitStyles?: IStyle;
+  subStyles?: IStyle;
+}
+
+/**
+ * A component to allow users to enter phone number through clicking on dialpad/using keyboard
+ *
+ *
+ * @public
+ */
+export const Dialpad = (props: DialpadProps): JSX.Element => {
+  const localeStrings = useLocale().strings.dialpad;
+  const strings = { ...localeStrings, ...props.strings };
+  return <DialpadContainer errorText={strings.errorText} defaultText={strings.defaultText} {...props} />;
+};
+
+const formatPhoneNumber = (value: String): string => {
+  // if input value is falsy eg if the user deletes the input, then just return
+  if (!value) {
+    return value;
+  }
+
+  // clean the input for any non-digit values.
+  let phoneNumber = value.replace(/[^\d*#+]/g, '');
+
+  // if phone number starts with 1, format like 1 (xxx)xxx-xxxx.
+  // if phone number starts with +, we format like +x (xxx)xxx-xxxx.
+  // For now we are only supporting NA phone number formatting with country code +x
+  // first we chop off the countrycode then we add it on when returning
+  let countryCodeNA = '';
+  if (phoneNumber[0] === '1') {
+    countryCodeNA = '1';
+
+    phoneNumber = phoneNumber.slice(1, phoneNumber.length);
+  } else if (phoneNumber[0] === '+') {
+    countryCodeNA = phoneNumber.slice(0, 2);
+    phoneNumber = phoneNumber.slice(2, phoneNumber.length);
+  }
+
+  // phoneNumberLength is used to know when to apply our formatting for the phone number
+  const phoneNumberLength = phoneNumber.length;
+
+  // we need to return the value with no formatting if its less then four digits
+  // this is to avoid weird behavior that occurs if you  format the area code too early
+  // if phoneNumberLength is greater than 10 we don't do any formatting
+
+  if (phoneNumberLength < 4 || phoneNumberLength > 10) {
+    return countryCodeNA + phoneNumber;
+  }
+
+  // if phoneNumberLength is greater than 4 and less the 7 we start to return
+  // the formatted number
+  if (phoneNumberLength < 7) {
+    return `${countryCodeNA}(${phoneNumber.slice(0, 3)})${phoneNumber.slice(3)}`;
```

Comment:
should there be a space after country code and after the first slice?:

| current | expected |
| -- | -- |
| +1(555)555-5555 | +1 (555) 555-5555 |

Reply:
updated